### PR TITLE
remove dependency on apiextensions

### PR DESCRIPTION
**Pull request by ubergesundheit**
This PR removes the dependency on apiextensions so it will be possible to add or remove fields to the release crd without updating this plugin

@giantswarm/team-ludacris could you approve? I was unable to add you to reviewers through github ui

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -8,32 +8,37 @@ import (
...
 	"strings"
 
-	"github.com/giantswarm/apiextensions/pkg/apis/release/v1alpha1"
 	"sigs.k8s.io/yaml"
 )
 
-func handleDocument(provider string, document []byte) {
-	var release v1alpha1.Release
+func handleDocument(provider, annotationKey string, document []byte) {
+	var release map[string]interface{}
```
Comment (tfussell) [inline main.go]: For a little more type safety, you could use `k8s.io/apimachinery/pkg/apis/meta/v1/unstructured.Unstructured` like this
```go
var release unstructured.Unstructured
err := yaml.UnmarshalStrict(document, &release)
if err != nil {
  // handle
}
annotations := release.GetAnnotations()
// maybe check for `nil` annotations
annotations["key"] = "value"
release.SetAnnotations(annotations)
modified, err := yaml.Marshal(release)
if err != nil {
  // handle
}
```

Comment (ubergesundheit) [reply main.go]: Thank you very much for your input. I've applied your suggestion in https://github.com/giantswarm/kustomize-plugin-releasenotesurlannotationtransformer/pull/2/commits/47a7c0809c80ae8edb7db90c4e550f947e1be2c7

File: main.go [hunk 2]
```diff
@@ -8,32 +8,42 @@ import (
...
 		fmt.Println(err)
-		os.Exit(4)
+		os.Exit(1)
 	}
 
-	if release.Kind == "Release" {
-		release.Annotations["giantswarm.io/release-notes"] = fmt.Sprintf("https://github.com/giantswarm/releases/tree/master/%s/%s", provider, release.Name)
+	if release.GetKind() == "Release" {
+		annotations := release.GetAnnotations()
+		if annotations == nil {
```
Comment (tfussell) [inline main.go]: We expect this to always have a default value so this is probably fine, but you could also just initialize the map if it's nil.

File: go.mod [hunk 3]
```diff
@@ -3,6 +3,6 @@ module github.com/giantswarm/kustomize-plugin-releasenotesurlannotationtransform
 go 1.14
 
 require (
-	github.com/giantswarm/apiextensions v0.4.14
+	k8s.io/apimachinery v0.18.6
```
Comment (tfussell) [inline go.mod]: I think we're targeting v1.18.5 for all of our repos.

Comment (tfussell) [reply go.mod]: https://github.com/giantswarm/giantswarm/blob/232aef0faa24cd0f44d0125da8f2061b5e3e427c/content/docs/dev-and-releng/client-go/k8s-1-18/_index.md

File: main.go (hunk 2 above)
Comment (ubergesundheit) [reply main.go]: If the map is nil then there are bigger problems in the releases repository I think. So I'll remove this check altogether..

File: go.mod (hunk 3 above)
Comment (ubergesundheit) [reply go.mod]: Do you know where this is documented?

Comment (tfussell) [reply go.mod]: Docs are still in progress, see the link above though.

Comment (ubergesundheit) [reply go.mod]: thanks! Downgraded to 1.18.5

**Review by MarcelMue — APPROVED** [review]

**Review by tfussell — APPROVED** [review]

**Pull request merged**